Issue #16: started configuring logback logging

artemlisikh opened the pull request:
but didn't figure out how to remove deprecated logging info yet

artemlisikh committed:
started configuring logging, but didn't figure out how to remove deprecated logging info yet

olisikh commented on src/main/resources/logback-spring.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
I propose to have logback.groovy instead of XML, XML is shit!

olisikh replied:
Nice that you're using spring boot integration for logging, let's stick to it. Here's a page, we need to add groovy dependency and migrate logback-spring.xml to logback.groovy

Here's a link: https://docs.spring.io/spring-boot/docs/current/reference/html/howto-logging.html#howto-configure-logback-for-logging

In groovy file, you'll be able to access System.getProperty("LOG_FILE") which would get you this system property, we don't want to use maven filtering here (@ logging.file @) as it's quite dangerous, we'll only use it for application.yml.

artemlisikh replied:
Great idea! I totally support that.

olisikh replied:
You're my star!

olisikh commented on pom.xml:
```diff
@@ -23,12 +23,13 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>1.8</java.version>
         <flyway.version>4.2.0</flyway.version>
+        <logging.file>logs/spring-boot-logging.log</logging.file>
```
let's name it ${project.name}.log, what do you think?

artemlisikh replied:
My thoughts exactly

artemlisikh committed:
replaced xml logback configuration to groovy
rollback now works just fine: every day logs are archived
archive directory's size can't be bigger than 10MB, so files in it will be deleted automatically when their time comes

artemlisikh committed:
changed log file's name

artemlisikh reviewed (commented):
replaced ... with *

Commit:
Merge remote-tracking branch 'origin/develop' into add_file_logging

# Conflicts:
#	pom.xml

artemlisikh committed:
great news, now path to logs can be dynamically changed

artemlisikh committed:
Merge remote-tracking branch 'origin/add_file_logging' into add_file_logging

artemlisikh commented:
what has just happened

olisikh commented on pom.xml:
```diff
@@ -29,6 +29,9 @@
         <db.password>5472</db.password>
         <spring.profiles.active>default</spring.profiles.active>
+
+        <logging.file>${project.name}.log</logging.file>
```
I think we need to have logging.file only as a ${project.name}. Why? See next comments for logbackx.groovy

olisikh commented on src/main/resources/logbackx.groovy:
```diff
@@ -0,0 +1,32 @@
+    file = "${LOG_PATH}/${LOG_FILE}"
+
+    rollingPolicy(TimeBasedRollingPolicy) {
+        fileNamePattern = "${LOG_ARCHIVE}/rollingfile.log%d{yyyy-MM-dd}.log"
```
I think we need to have a name of the file as "${LOG_ARCHIVE}/${LOG_FILE}-%d{yyyy-MM-dd}.log" and this is why I think we should have LOG_FILE without extension.

konteno replied:
Can't apply that pattern because there's no filenamepattern for log file, only for archived log files

olisikh commented on src/main/resources/logbackx.groovy:
```diff
@@ -0,0 +1,32 @@
+}
+
+appender("FILE", RollingFileAppender) {
+    file = "${LOG_PATH}/${LOG_FILE}"
```
If we change LOG_FILE to be extensionless, we need to add .log to the end of this string.

artemlisikh committed:
changed archive filename pattern

olisikh reviewed (approved):
Brilliant! Great job!
